### ISSUE TITLE
docker-compose with version 2 enabled doesn't pass correct arguments to psql binary

### ISSUE THREAD

**Issue by eddorre** [docker/compose-cli#1962]
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------

**Description**

Running `docker-compose` with v2 enabled doesn't seem to pass the correct arguments to the psql binary.

**Steps to reproduce the issue:**
Create a simple docker-compose.yml file:

```
services:
  app:
    build:
      context: .
  
  postgres:
    image: postgis/postgis:11-2.5
    volumes:
      - postgres-data-test:/var/lib/postgresql/data
    ports:
      - 5342:5432
    environment:
      POSTGRES_DB: xbuild
      POSTGRES_HOST_AUTH_METHOD: trust

volumes:
  postgres-data-test:
```

Create a simple Dockerfile:

```
FROM alpine:3.14
```

Run `docker-compose enable-v2`
Run `docker build`
Run `docker-compose up -d`
Run `docker-compose exec postgres psql -h postgres -U postgres -c 'create database xbuild'`.

**Describe the results you received:**

The `psql` binary outputs usage information.

Sample output:

```
docker-compose exec postgres psql -h postgres -U postgres -c 'create database xbuild'
psql is the PostgreSQL interactive terminal.
Usage:
  psql [OPTION]... [DBNAME [USERNAME]]
General options:
  -c, --command=COMMAND    run only single command (SQL or internal) and exit
  -d, --dbname=DBNAME      database name to connect to (default: "root")
  -f, --file=FILENAME      execute commands from file, then exit
  -l, --list               list available databases, then exit
  -v, --set=, --variable=NAME=VALUE
                           set psql variable NAME to VALUE
                           (e.g., -v ON_ERROR_STOP=1)
  -V, --version            output version information, then exit
  -X, --no-psqlrc          do not read startup file (~/.psqlrc)
  -1 ("one"), --single-transaction
                           execute as a single transaction (if non-interactive)
  -?, --help[=options]     show this help, then exit
      --help=commands      list backslash commands, then exit
      --help=variables     list special variables, then exit
Input and output options:
  -a, --echo-all           echo all input from script
  -b, --echo-errors        echo failed commands
  -e, --echo-queries       echo commands sent to server
  -E, --echo-hidden        display queries that internal commands generate
  -L, --log-file=FILENAME  send session log to file
  -n, --no-readline        disable enhanced command line editing (readline)
  -o, --output=FILENAME    send query results to file (or |pipe)
  -q, --quiet              run quietly (no messages, only query output)
  -s, --single-step        single-step mode (confirm each query)
  -S, --single-line        single-line mode (end of line terminates SQL command)
Output format options:
  -A, --no-align           unaligned table output mode
  -F, --field-separator=STRING
                           field separator for unaligned output (default: "|")
  -H, --html               HTML table output mode
  -P, --pset=VAR[=ARG]     set printing option VAR to ARG (see \pset command)
  -R, --record-separator=STRING
                           record separator for unaligned output (default: newline)
  -t, --tuples-only        print rows only
  -T, --table-attr=TEXT    set HTML table tag attributes (e.g., width, border)
  -x, --expanded           turn on expanded table output
  -z, --field-separator-zero
                           set field separator for unaligned output to zero byte
  -0, --record-separator-zero
                           set record separator for unaligned output to zero byte
Connection options:
  -h, --host=HOSTNAME      database server host or socket directory (default: "local socket")
  -p, --port=PORT          database server port (default: "5432")
  -U, --username=USERNAME  database user name (default: "root")
  -w, --no-password        never prompt for password
  -W, --password           force password prompt (should happen automatically)
For more information, type "\?" (for internal commands) or "\help" (for SQL
commands) from within psql, or consult the psql section in the PostgreSQL
documentation.
Report bugs to <pgsql-bugs@postgresql.org>.
```

**Describe the results you expected:**

Running `docker-compose exec postgres psql -h postgres -U postgres -c 'create database xbuild'` should execute successfully.

**Additional information you deem important (e.g. issue happens only occasionally):**

If v2 is disabled then the command runs successfully. 
If one runs the command with all the command line parameters without shortcuts it works: `docker-compose exec postgres psql --host=postgres --username=postgres --command='create database xbuild'`
If one runs the command using docker compose it runs successfully: `docker compose exec postgres psql -h postgres -U postgres -c 'create database xbuild'`

**Output of `docker-compose version`:**
```
docker-compose version 1.29.2, build 5becea4c
docker-py version: 5.0.0
CPython version: 3.9.0
OpenSSL version: OpenSSL 1.1.1h  22 Sep 2020
```

**Output of `docker compose version`:**
```
Docker Compose version v2.0.0-beta.6
```

**Output of `docker version`:**

```
Client:
 Cloud integration: 1.0.17
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.16.4
 Git commit:        f0df350
 Built:             Wed Jun  2 11:56:22 2021
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true
Server: Docker Engine - Community
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       b0f5bc3
  Built:            Wed Jun  2 11:54:58 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.6
  GitCommit:        d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc:
  Version:          1.0.0-rc95
  GitCommit:        b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker context show`:**  
You can also run `docker context inspect context-name` to give us more details but don't forget to remove sensitive content.

```
default
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
  compose: Docker Compose (Docker Inc., v2.0.0-beta.6)
  scan: Docker Scan (Docker Inc., v0.8.0)
Server:
 Containers: 2
  Running: 1
  Paused: 0
  Stopped: 1
 Images: 6
 Server Version: 20.10.7
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc version: b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.25-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 11.7GiB
 Name: docker-desktop
 ID: V3DX:AUF2:EPMK:TDUS:DSMG:Y7BT:IUGQ:DEBJ:VM5Z:HC6T:JFGJ:N3GY
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details (AWS ECS, Azure ACI, local, etc.):**
Local

**Comment by ndeloof** [comment]
Tried to reproduce, don't get this issue:
```console
$ compose exec postgres psql -h postgres -U postgres -c 'create database xbuild'
ERROR:  database "xbuild" already exists
```

**Comment by ndeloof** [comment]
Confirmed use of the `docker-compose` compatibility wrapper is responsible for this issue.

**Comment by ndeloof** [comment]
Internally tracked as https://docker.atlassian.net/browse/IL-479

**Comment by erwittry** [comment]
Is there any update on when this issue will be resolved?

**Comment by ndeloof** [comment]
Fix will be released with Docker Desktop 4.1.0 to be released end of September
